# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/acl/install.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./install-sh.orig	2002-06-05 01:07:56.000000000 +0200
+++ ./install-sh	2003-04-15 07:42:07.000000000 +0200
@@ -122,9 +122,9 @@
 
 if $INSTALL
 then
-    CP=cp; LN=ln; MKDIR=mkdir; CHMOD=chmod; CHOWN=_chown
+    MV=mv; CP=cp; LN=ln; MKDIR=mkdir; CHMOD=chmod; CHOWN=_chown
 else
-    CP=true; LN=true; MKDIR=true; CHMOD=true; CHOWN=true
+    MV=mv; CP=true; LN=true; MKDIR=true; CHMOD=true; CHOWN=true
 fi
 
 [ -n "$DIST_ROOT" -a $REAL_UID -ne 0 ] && CHOWN=true
@@ -229,7 +229,8 @@
 		if [ $cnt -eq 5 ]
 		then
 			install_name=$target/$solib
-			$CP $solib $install_name
+			$CP $solib $install_name.new
+			$MV $install_name.new $install_name
 			status=$?
 			$MANIFEST f $FILEMODE $OWNER $GROUP $HERE/$solib ${install_name#$DIST_ROOT}
 			break
@@ -280,7 +281,8 @@
 	;;
     old_lib)
 	install_name=$target/$old_library
-	$CP $old_library $install_name
+	$CP $old_library $install_name.new
+	$MV $install_name.new $install_name
 	status=$?
 	$MANIFEST f $FILEMODE $OWNER $GROUP $HERE/$old_library ${install_name#$DIST_ROOT}
 	;;
@@ -314,7 +316,13 @@
 	then
 	    mkdir -p `dirname $dir`
 	fi
-	$CP $f $dir
+	if test -d $dir; then
+	    $CP $f $dir/${f##*/}.new
+	    $MV $dir/${f##*/}.new $dir/${f##*/}
+	else
+	    $CP $f $dir.new
+	    $MV $dir.new $dir
+	fi
 	status=$?
 	if [ $status -eq 0 ]
 	then 
@@ -357,7 +365,13 @@
 
 	for f in $list
 	do
-	    $CP $f $dir
+	    if test -d $dir; then
+		$CP $f $dir/${f##*/}.new
+		$MV $dir/${f##*/}.new $dir/${f##*/}
+	    else
+		$CP $f $dir.new
+		$MV $dir.new $dir
+	    fi
 	    status=$?
 	    if [ $status -eq 0 ]
 	    then
